blog/_posts/2021-01-19-scala-native-0.4-release.md

```diff
@@ -0,0 +1,225 @@
+
+It would not be wise to enable reflective calls to all definitions as we would need to load many more objects and increase the size of the binary.
+That’s why you need to signal to the compiler which classes can use this mechanism - it can be done by marking your `trait`, `class`, or `object`
+with the `@EnableReflectiveInstantation` annotation. It will also make all of its descendants able to be reflectively instantiated.
+
+Annotated classes and modules, having a concrete implementation, can be accessed via the provided `scalanative.reflect.Reflect` API.
+If you have used Scala.js before, it may seem similar to you, as the new implementation uses exactly the same API.
+
+## Cross-platform testing using JUnit
+Until recently, all our internal tests were executed using our own native testing framework, mainly due to lack of proper
+reflection support in Scala Native. After getting rid of this blocker, we have decided to use JUnit instead and in 
+result implemented the native plugin for this framework. Thanks to this change you are also able to also use it in your own projects.  
+Naturally you are still able to use any other testing framework, as long as it supports Scala Native.
+
+To enable Native JUnit plugin tests, all you need to do is to add the two following lines to your `build.sbt`.
+```scala
+addCompilerPlugin("org.scala-native" % "junit-plugin" % nativeVersion cross CrossVersion.full)
+libraryDependencies += "org.scala-native" %%% "junit-runtime" % nativeVersion % "test"
+```
+
+The above settings, when running `sbt test`, would result in the tests being compiled using the SN plugin and executed 
+with a native implementation of the JUnit framework. What is more you can easily run your tests both on JVM and Native 
+with a simple adjustment to your build, you can find guide how to do this in [Scala Native reference](https://scala-native.readthedocs.io/en/latest/)
+
+Unfortunately, SN is still single-threaded, so if your tests are using concurrency you will need to add the `junit-async`
+runtime to your build. It would use the default `ExecutionContext` on the JVM and it would mock concurrent execution in an
+event loop in Native - each `Runnable` would be run in order of invocation after finishing the main thread.
+
+```scala
+// for JVM submodule
+libraryDependencies += "org.scala-native" %% "junit-async" % SNVersion
+
+// for Native submodule
+libraryDependencies += "org.scala-native" %%% "junit-async-native" % SNVersion
+```
+
+> Scala Native assumes single-threaded execution by default and does not provide multi-threaded programming, 
+> however it’s possible to use C libraries to get access to multi-threading and synchronization primitives. 
+> You need to remember it is not officially supported at the moment.
+
+## Interop changes
+### Including Native Code in your Application or Library
+Previously, Scala Native used C source files in its build pipeline allowing to mix Scala code with native libraries,
```

Comment:
"allowing to mix" isn't grammatical. perhaps "allowing you to mix" or "allowing mixing" or "so you can mix"